libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/detail/search.hpp

```diff
@@ -0,0 +1,179 @@
+//  Copyright (C) ETH Zurich
+//  Copyright (C) Eric Niebler 2014-2020
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+//===----------------------------------------------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is dual licensed under the MIT and the University of Illinois Open
+// Source Licenses. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
```

Comment:
Are we allowed to re-license this under the Boost license?

Reply:
So I checked that the MIT license allows us to "use, copy, modify, merge, publish, distribute", do I need to check specifically something else?

Reply:
It also requires us to redistribute the original license, iiuc. I would suggest to avoid directly using code that originally was not published under the BSL. What part of the file did you take from LLVM?

Reply:
Ah okay didn't know that exactly, I took the sequential impl from the range-v3 repo.
I saw some "possible implementations" on cppreference but it is probably a bit slower, what do you suggest?

Reply:
Would putting the original MIT license on top of the header taking only the range-v3 sequential adaptation be enough?

Reply:
There are at least two ways out. Either - as you suggested:
- use the sequential implementation from cppreference and adapt it to support projections
- use the `std::search` as before, but pass our `compare_projected` helper function object 

Both are fine as a starting point, I believe.

Reply:
BTW, once you have removed the license, please force-push to this branch such that the commit completely disappears from the git history.

Reply:
Yes no problem :+1: